src/process_utility.c

```diff
@@ -2563,19 +2566,50 @@ process_altertable_validate_constraint_end(Hypertable *ht, AlterTableCmd *cmd)
 }
 
 static void
-process_altertable_drop_not_null(Hypertable *ht, AlterTableCmd *cmd)
+validate_uses_hypercore(Hypertable *ht, Oid chunk_relid, void *arg)
```

Comment:
Slightly misleading name of the function.

You are actually checking two things and failing only if:

- The command is a SetNotNull
- The table is a "regular" compressed chunk

So, the check will pass for anything else, including non-compressed chunks and other TAMs. But the function seems to give the impression that the table is using Hypercore if the validation succeeds, which is not true in all cases.

The old style function name is probably better if changed to _set_not_null.